feat: add new multichain account version for sitecell

montelaidev opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR adds two new ui components. `MultichainAccountSiteCell` and `MultichainAccountSiteCellToolTip` that uses the new account groups. The PR does not hook up the components and are not testable. 

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/35302?quickstart=1)

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

Not testable because it is not used. 

## **Screenshots/Recordings**

### **After**

These are storybook screenshots.
<img width="234" height="216" alt="Screenshot 2025-08-21 at 21 12 56" src="https://github.com/user-attachments/assets/d37ab46f-83d4-4dda-823e-96b46c970f1a" />
<img width="426" height="213" alt="Screenshot 2025-08-21 at 21 20 07" src="https://github.com/user-attachments/assets/52ecb16f-f545-4a85-9972-c2599006a656" />



<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

gantunesr reviewed (commented):
looks good, small question left

montelaidev commented on ui/components/multichain-accounts/permissions/edit-accounts-modal/multichain-edit-accounts-modal.tsx:
```diff
@@ -0,0 +1,157 @@
+    setSelectedAccountGroups(defaultSelectedAccountGroups);
+  }, [
+    // TODO: Fix the source of this prop value to be the same array instance each render
+    JSON.stringify(defaultSelectedAccountGroups),
```
This can be removed

montelaidev committed:
feat: multichain version of site cell

montelaidev committed:
fix: lint

montelaidev committed:
fix: test and lint

montelaidev committed:
fix: lint

montelaidev committed:
fix: tests

montelaidev committed:
fix: lint and remove story that belongs to subsequent pr

PatrykLucka commented on ui/components/multichain-accounts/permissions/multichain-edit-networks-page/multichain-edit-networks-page.tsx:
```diff
@@ -0,0 +1,239 @@
+  );
+
+  const selectAll = () => {
+    const allNetworksChainIds = allNetworks.map(
```
I think we could move it outside of this function with useMemo, so we don't have to run in every time selectAll is clicked.

montelaidev committed:
refactor: optimize state management in MultichainEditNetworksPage

- Replaced direct array mapping with useMemo for allNetworksChainIds to improve performance.
- Utilized useCallback for selectAll and handleNetworkClick functions to prevent unnecessary re-renders.
- Updated allAreSelected to useMemo for better efficiency in determining selection state.

montelaidev committed:
fix: lint
